### ISSUE TITLE
unused values in argument list should generate error.

### ISSUE THREAD

**Issue by ScottBailey** [AntelopeIO/DUNE#69]
The following commands should fail, instead they work in ways the user might not intend :
```
./dune --start my_node my_config.ini  # the user meant to type '--start my_node --config my_config.ini'
./dune xxyyzzy  # nothing happens, an error should be displayed with usage.
```
Essentially, each argument needs to evaluate if the `remainder` value makes sense and needs to act on it.

**Comment by stephenpdeos** [comment]
@ScottBailey to verify by EOD whether this should be closed or not. Let's link a PR if we have a relevant one to attach. Thanks!

**Comment by ScottBailey** [comment]
I have confirmed this issue remains in db7bb0606235a5a57a050ba3d6e79f740e506ac2

**Comment by mikelik** [comment]
Fixed in #117
